src/Talis/Persona/Client/Login.php

```diff
@@ -61,9 +61,19 @@ public function requireAuth($provider, $appId, $appSecret, $redirectUri = '')
      */
     public function validateAuth()
     {
+        if (isset($_POST['persona:signature']))
+        {
+            $payloadSignature = $_POST['persona:signature'];
+        } else {
+            $this->getLogger()->error("Signature not set");
+            throw new \Exception('Signature not set');
```

Comment:
In my opinion, if you can't validate the payload because the signature is missing, it's insecure so a 401 is the right thing to return. Let's not worry about it for this change because I think you should use this in Depot first to get a feel for things.